File: Examples/Example-Java/app/src/main/java/com/zettle/payments/android/java_example/TippingStyleBottomSheet.java
package com.zettle.payments.android.java_example;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
import com.zettle.sdk.feature.cardreader.payment.TippingStyle;

public class TippingStyleBottomSheet extends BottomSheetDialogFragment {

    private View styleNoneView;
    private View styleDefaultView;
    private View styleAmountView;
    private View stylePercentView;

    public static final String REQUEST_KEY = "TippingStyleBottomSheetRC";
    public static final String TIPPING_STYLE_KEY = "TippingStyleKey";

    public static TippingStyleBottomSheet newInstance() {
        return new TippingStyleBottomSheet();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_tipping_style, container, false);
        styleNoneView = view.findViewById(R.id.tipping_style_none);
        styleDefaultView = view.findViewById(R.id.tipping_style_default);
        styleAmountView = view.findViewById(R.id.tipping_style_amount);
        stylePercentView = view.findViewById(R.id.tipping_style_percent);

        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        styleNoneView.setOnClickListener(v -> setResultAndExit(TippingStyle.None));
        styleDefaultView.setOnClickListener(v -> setResultAndExit(TippingStyle.Default));
        styleAmountView.setOnClickListener(v -> setResultAndExit(TippingStyle.Amount));
        stylePercentView.setOnClickListener(v -> setResultAndExit(TippingStyle.Percentage));
    }

    private void setResultAndExit(TippingStyle tippingStyle) {
        Bundle bundle = new Bundle();
        bundle.putSerializable(TIPPING_STYLE_KEY, tippingStyle);

        getParentFragmentManager().setFragmentResult(REQUEST_KEY, bundle);
        dismiss();
    }
}

File: Examples/Example-Kotlin/app/src/main/java/com/zettle/payments/android/kotlin_example/TippingStyleBottomSheet.kt
package com.zettle.payments.android.kotlin_example

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.setFragmentResult
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import com.zettle.sdk.feature.cardreader.payment.ZettleReaderTippingStyle

class TippingStyleBottomSheet : BottomSheetDialogFragment() {

    private lateinit var styleNoneView: View
    private lateinit var styleDefaultView: View
    private lateinit var styleAmountView: View
    private lateinit var stylePercentView: View

    companion object {
        const val REQUEST_KEY = "TippingStyleBottomSheetRC"
        const val TIPPING_STYLE_KEY = "TippingStyleKey"

        fun newInstance() = TippingStyleBottomSheet()
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val binding = inflater.inflate(R.layout.fragment_tipping_style, container, false)
        styleNoneView = binding.findViewById(R.id.tipping_style_none)
        styleDefaultView = binding.findViewById(R.id.tipping_style_default)
        styleAmountView = binding.findViewById(R.id.tipping_style_amount)
        stylePercentView = binding.findViewById(R.id.tipping_style_percent)

        return binding.rootView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        styleNoneView.setOnClickListener { setResultAndExit(ZettleReaderTippingStyle.None) }
        styleDefaultView.setOnClickListener { setResultAndExit(ZettleReaderTippingStyle.Default) }
        styleAmountView.setOnClickListener { setResultAndExit(ZettleReaderTippingStyle.Amount) }
        stylePercentView.setOnClickListener { setResultAndExit(ZettleReaderTippingStyle.Percentage) }
    }

    private fun setResultAndExit(tippingStyle: ZettleReaderTippingStyle) {
        setFragmentResult(REQUEST_KEY, Bundle().apply { putSerializable(TIPPING_STYLE_KEY, tippingStyle) })
        dismiss()
    }
}